test/runner.js

```diff
@@ -327,6 +338,19 @@ describe('Runner', function(){
       fail.should.throw('allow unhandled errors');
       done();
     });
+
+    it('should allow unhandled errors in hooks to propagate through', function(done) {
```

Comment:
@boneskull  I'm sorry for delay.

I can not think of a good way, how to check the exception handling asynchronously. Perhaps I not fully understand the problem. Please give me an advice on how to do it correctly.